[Alerts table] Edit tags UI

benakansara opened the pull request:
## Release Notes

Users can now manually add workflow tags directly to alerts (see [this PR as well](https://github.com/elastic/kibana/pull/244251))


## Summary 

Resolves https://github.com/elastic/kibana/issues/240376

- Adds "Edit tags" in alert actions
- Opens flyout when clicking on "Edit tags"
- Flyout shows list of all `kibana.alert.workflow_tags` from the selected alerts
- "Edit tags" will be disabled when all alerts are selected (as there can be 1000s of alerts with many tags)
- "Edit tags" will not be available for Security rules

benakansara committed:
init commit

benakansara committed:
show edit tags flyout

benakansara committed:
add tests

benakansara committed:
cleanup

benakansara committed:
Merge branch 'main' into feat/edit-workflow-tags

benakansara committed:
do not show edit tags for sec rules

benakansara committed:
fix ci errors

benakansara committed:
fix tests

benakansara committed:
Merge branch 'main' into feat/edit-workflow-tags

benakansara committed:
refactoring

benakansara committed:
refactoring

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/363952)
* Commit: 18d37d14da368ae1f0ddcbfd39d44d9f7d6a0e67

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7)
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379)
* [FTR Configs #53](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad5e-417f-8ec7-01562c6b47f0)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa831-ad21-4521-a065-150a5cda51f7/artifacts/019aa840-8205-47f4-9a85-4c668fe864b7) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions column when the case service is defined and the user does not have read access
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa840-9715-44ba-8ab3-e80e61343379/artifacts/019aa850-2eb9-45ac-a1e2-b913dfd78ecc) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions column when the case service is defined and the user does not have read access
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa831-ad21-4521-a065-150a5cda51f7/artifacts/019aa840-8205-4776-85b8-941e4609a188) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions column when the case service is defined and the user does not have write access
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa840-9715-44ba-8ab3-e80e61343379/artifacts/019aa850-2eb9-4264-8349-0ac8211f4e20) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions column when the case service is defined and the user does not have write access
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa831-ad21-4521-a065-150a5cda51f7/artifacts/019aa840-8205-4282-af49-ba598058f275) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions when the cases context is missing
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa840-9715-44ba-8ab3-e80e61343379/artifacts/019aa850-2eba-4f94-863e-25e29da0c7f5) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions Cases should not show the bulk actions when the cases context is missing
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa831-ad21-4521-a065-150a5cda51f7/artifacts/019aa840-8204-4403-bb87-71733bb85ef5) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions when the additionalBulkActions option is not set should not show the bulk actions column
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa840-9715-44ba-8ab3-e80e61343379/artifacts/019aa850-2eb8-4199-b16b-8535c515d098) Jest Tests #<span></span>6 / AlertsDataGrid bulk actions when the additionalBulkActions option is not set should not show the bulk actions column
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa831-ad21-4521-a065-150a5cda51f7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa831-ad21-4521-a065-150a5cda51f7/artifacts/019aa840-8206-4946-9abc-d2633e1900aa) Jest Tests #<span></span>6 / bulk action hooks useBulkActions appends the case and untrack bulk actions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/363952#019aa840-9715-44ba-8ab3-e80e61343379) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/363952/jobs/019aa840-9715-44ba-8ab3-e80e61343379/artifacts/019aa850-2eba-4c44-a2f2-d2fc983325ed) Jest Tests #<span></span>6 / bulk action hooks useBulkActions appends the case and untrack bulk actions

### History
* :broken_heart: [Build #363693](https://buildkite.com/elastic/kibana-pull-request/builds/363693) failed f813798b62d0679d704e252dc6a7521e130784fd

cc @benakansara

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/363952","number":363952,"commit":"18d37d14da368ae1f0ddcbfd39d44d9f7d6a0e67"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/363693","number":363693,"commit":"f813798b62d0679d704e252dc6a7521e130784fd"}],"number":363952}
buildkite-pr-comment-->

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

baileycash-elastic committed:
integrate with api

I removed useItemsAction here because that hook was designed for the alert update api, which does not easily support updating workflow tags. I moved the logic to useTagsAction instead. useItemsAction may come in use in the future if we decide to directly write to the tags field of an alert.

baileycash-elastic committed:
hook cleanup, fix tests

other improvements

baileycash-elastic commented:
I picked this up earlier today. 

Note: I removed useItemsAction here because that hook was designed for the alert update api, which does not easily support updating workflow tags. I moved the logic to useTagsAction instead. useItemsAction may come in use in the future if we decide to directly write to the tags field of an alert.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/365127)
* Commit: 13c0b952f6eddb22c3ff39bf9b9983e61a29cc19

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/907e877fadd6acd3c7f115bfea0f8296bcf01147) | [after](https://github.com/elastic/kibana/commit/13c0b952f6eddb22c3ff39bf9b9983e61a29cc19) | diff |
| --- | --- | --- | --- |
| `cases` | 1191 | 1198 | +7 |
| `embeddableAlertsTable` | 561 | 569 | +8 |
| `ml` | 2811 | 2819 | +8 |
| `observability` | 1598 | 1606 | +8 |
| `securitySolution` | 8414 | 8421 | +7 |
| `triggersActionsUi` | 1091 | 1099 | +8 |
| total |  |  | +46 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/907e877fadd6acd3c7f115bfea0f8296bcf01147) | [after](https://github.com/elastic/kibana/commit/13c0b952f6eddb22c3ff39bf9b9983e61a29cc19) | diff |
| --- | --- | --- | --- |
| `cases` | 1.4MB | 1.4MB | +12.0KB |
| `embeddableAlertsTable` | 1000.7KB | 1013.6KB | +12.8KB |
| `ml` | 5.6MB | 5.6MB | +12.8KB |
| `observability` | 1.7MB | 1.7MB | +13.2KB |
| `securitySolution` | 11.1MB | 11.1MB | +11.9KB |
| `triggersActionsUi` | 1.5MB | 1.5MB | +13.2KB |
| total |  |  | +75.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/907e877fadd6acd3c7f115bfea0f8296bcf01147) | [after](https://github.com/elastic/kibana/commit/13c0b952f6eddb22c3ff39bf9b9983e61a29cc19) | diff |
| --- | --- | --- | --- |
| `observability` | 95.9KB | 95.9KB | -1.0B |



### History
* :broken_heart: [Build #364681](https://buildkite.com/elastic/kibana-pull-request/builds/364681) failed f039b462e2429828e6380989625d570dbc570a72
* :broken_heart: [Build #364460](https://buildkite.com/elastic/kibana-pull-request/builds/364460) failed b81e7ee0a4476b43062bde6f3245689f6d052795
* :broken_heart: [Build #363952](https://buildkite.com/elastic/kibana-pull-request/builds/363952) failed 18d37d14da368ae1f0ddcbfd39d44d9f7d6a0e67
* :broken_heart: [Build #363693](https://buildkite.com/elastic/kibana-pull-request/builds/363693) failed f813798b62d0679d704e252dc6a7521e130784fd

cc @benakansara @baileycash-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365127","number":365127,"commit":"13c0b952f6eddb22c3ff39bf9b9983e61a29cc19"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/364681","number":364681,"commit":"f039b462e2429828e6380989625d570dbc570a72"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/364460","number":364460,"commit":"b81e7ee0a4476b43062bde6f3245689f6d052795"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/363952","number":363952,"commit":"18d37d14da368ae1f0ddcbfd39d44d9f7d6a0e67"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/363693","number":363693,"commit":"f813798b62d0679d704e252dc6a7521e130784fd"}],"number":365127}
buildkite-pr-comment-->

baileycash-elastic committed:
fix typing compile issue

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

cnasikas commented on src/platform/packages/shared/response-ops/alerts-table/components/tags/edit_tags_flyout.test.tsx:
```diff
@@ -0,0 +1,88 @@
+  it('shows the number of total selected alerts in the title when selecting multiple alerts', async () => {
+    const mockAlert2 = {
+      ...mockAlert,
+      id: 'alert-2',
```
```suggestion
      _id: 'alert-2',
```

cnasikas commented on src/platform/packages/shared/response-ops/alerts-table/hooks/use_bulk_update_alert_tags.tsx:
Let's add a small unit test for this hook.

cnasikas commented on src/platform/packages/shared/response-ops/alerts-table/reducers/bulk_actions_reducer.test.tsx:
What changed in the code and we had to change the tests to include `ruleTypeIds={['siem.esqlRule']}`?

cnasikas reviewed (commented):
LGTM! I left some comments. I also found a bug where the row action popover stays and does not hides when the flyout opens.

<img width="1479" height="959" alt="Image" src="https://github.com/user-attachments/assets/190d760c-3dcf-4ffa-b335-8be70e5be7e6" />

baileycash-elastic committed:
Update src/platform/packages/shared/response-ops/alerts-table/components/tags/edit_tags_flyout.test.tsx

Co-authored-by: Christos Nasikas <xristosnasikas@gmail.com>

baileycash-elastic committed:
Update src/platform/packages/shared/response-ops/alerts-table/components/tags/edit_tags_flyout.test.tsx

Co-authored-by: Christos Nasikas <xristosnasikas@gmail.com>

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

baileycash-elastic committed:
add tests

baileycash-elastic committed:
add context to handle flyout state

fix tests related to flyout changes

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

baileycash-elastic committed:
add more tests for alert grid

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

baileycash-elastic committed:
general cleanup

baileycash-elastic committed:
Merge remote-tracking branch 'refs/remotes/benakansara/feat/edit-workflow-tags' into feat/edit-workflow-tags

baileycash-elastic committed:
consolidate hook functionality

baileycash-elastic committed:
use generic index to update alerts

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

baileycash-elastic commented:
/ci

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368009)
* Commit: f7b623310949af3d8a9f43ade6e22f9cb383781e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e61bf7c6d8e4cb4f9209447b67dcfe8a686375c4) | [after](https://github.com/elastic/kibana/commit/f7b623310949af3d8a9f43ade6e22f9cb383781e) | diff |
| --- | --- | --- | --- |
| `cases` | 1195 | 1203 | +8 |
| `embeddableAlertsTable` | 562 | 571 | +9 |
| `ml` | 2816 | 2825 | +9 |
| `observability` | 1617 | 1626 | +9 |
| `securitySolution` | 8447 | 8455 | +8 |
| `triggersActionsUi` | 1093 | 1102 | +9 |
| total |  |  | +52 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e61bf7c6d8e4cb4f9209447b67dcfe8a686375c4) | [after](https://github.com/elastic/kibana/commit/f7b623310949af3d8a9f43ade6e22f9cb383781e) | diff |
| --- | --- | --- | --- |
| `cases` | 1.4MB | 1.4MB | +12.1KB |
| `embeddableAlertsTable` | 1000.8KB | 1013.4KB | +12.7KB |
| `ml` | 5.6MB | 5.6MB | +12.6KB |
| `observability` | 1.7MB | 1.7MB | +12.8KB |
| `securitySolution` | 11.1MB | 11.1MB | +12.1KB |
| `triggersActionsUi` | 1.5MB | 1.5MB | +12.8KB |
| total |  |  | +75.1KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e61bf7c6d8e4cb4f9209447b67dcfe8a686375c4) | [after](https://github.com/elastic/kibana/commit/f7b623310949af3d8a9f43ade6e22f9cb383781e) | diff |
| --- | --- | --- | --- |
| `embeddableAlertsTable` | 5.8KB | 5.8KB | +1.0B |
| `observability` | 95.7KB | 95.7KB | -1.0B |
| total |  |  | -0.0B |



### History
* :broken_heart: [Build #367443](https://buildkite.com/elastic/kibana-pull-request/builds/367443) failed 1e0f8454e3833dd6d026eff222460feb368ea368
* :yellow_heart: [Build #367228](https://buildkite.com/elastic/kibana-pull-request/builds/367228) was flaky 37cefd867bf2292b79d92e9ffc2f0e99acbd256a

cc @benakansara @baileycash-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368009","number":368009,"commit":"f7b623310949af3d8a9f43ade6e22f9cb383781e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367443","number":367443,"commit":"1e0f8454e3833dd6d026eff222460feb368ea368"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367228","number":367228,"commit":"37cefd867bf2292b79d92e9ffc2f0e99acbd256a"}],"number":368009}
buildkite-pr-comment-->

baileycash-elastic committed:
fix nits, add api assertions to tag action tests

baileycash-elastic committed:
add capabilities check to tags

baileycash-elastic committed:
add tests for edit action

baileycash-elastic commented on src/platform/packages/shared/response-ops/alerts-table/components/edit_tags_action.tsx:
[done](https://github.com/benakansara/kibana/commit/8569255d0ae6c3785a39a96cddd543fe1734a744)

cnasikas commented on src/platform/packages/shared/response-ops/alerts-table/hooks/use_bulk_update_alert_tags.test.tsx:
```diff
@@ -0,0 +1,281 @@
+    expect(onError).not.toHaveBeenCalled();
+  });
+
+  it('should update loading state during mutation', async () => {
```
Thank you so much for this great coverage! 

nit: Maybe we can remove this test as it tests React query and there is no custom logic regarding the `isLoading` and `isSuccess` booleans.

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#9973](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9973)
[❌] x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group4/config.ts: 6/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F243792%2Fhead)

baileycash-elastic committed:
attempt to fix flaky test

baileycash-elastic committed:
other flaky test fix

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368719)
* Commit: 3d0b25be9b00bbdf0eb688db4ec5337d367b3023

### Failed CI Steps
* [FTR Configs #15](https://buildkite.com/elastic/kibana-pull-request/builds/368719#019ae643-21a7-439c-9492-8a49375d410a)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/3b236efdc1a17b04beabf0f4250299fdd5fe7e37) | [after](https://github.com/elastic/kibana/commit/3d0b25be9b00bbdf0eb688db4ec5337d367b3023) | diff |
| --- | --- | --- | --- |
| `cases` | 1198 | 1206 | +8 |
| `embeddableAlertsTable` | 564 | 573 | +9 |
| `ml` | 2818 | 2827 | +9 |
| `observability` | 1618 | 1627 | +9 |
| `securitySolution` | 8455 | 8463 | +8 |
| `triggersActionsUi` | 1094 | 1103 | +9 |
| total |  |  | +52 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/3b236efdc1a17b04beabf0f4250299fdd5fe7e37) | [after](https://github.com/elastic/kibana/commit/3d0b25be9b00bbdf0eb688db4ec5337d367b3023) | diff |
| --- | --- | --- | --- |
| `cases` | 1.4MB | 1.4MB | +12.1KB |
| `embeddableAlertsTable` | 1001.0KB | 1013.7KB | +12.7KB |
| `ml` | 5.6MB | 5.6MB | +12.6KB |
| `observability` | 1.7MB | 1.7MB | +12.8KB |
| `securitySolution` | 11.1MB | 11.1MB | +12.1KB |
| `triggersActionsUi` | 1.5MB | 1.5MB | +12.8KB |
| total |  |  | +75.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/3b236efdc1a17b04beabf0f4250299fdd5fe7e37) | [after](https://github.com/elastic/kibana/commit/3d0b25be9b00bbdf0eb688db4ec5337d367b3023) | diff |
| --- | --- | --- | --- |
| `embeddableAlertsTable` | 5.8KB | 5.8KB | +1.0B |
| `observability` | 95.7KB | 95.7KB | -1.0B |
| total |  |  | -0.0B |



### History
* :broken_heart: [Build #368673](https://buildkite.com/elastic/kibana-pull-request/builds/368673) failed b13a985689f8021310e19a94e7e8a5340263651b
* :yellow_heart: [Build #368054](https://buildkite.com/elastic/kibana-pull-request/builds/368054) was flaky 8569255d0ae6c3785a39a96cddd543fe1734a744
* :green_heart: [Build #368009](https://buildkite.com/elastic/kibana-pull-request/builds/368009) succeeded f7b623310949af3d8a9f43ade6e22f9cb383781e
* :broken_heart: [Build #367443](https://buildkite.com/elastic/kibana-pull-request/builds/367443) failed 1e0f8454e3833dd6d026eff222460feb368ea368
* :yellow_heart: [Build #367228](https://buildkite.com/elastic/kibana-pull-request/builds/367228) was flaky 37cefd867bf2292b79d92e9ffc2f0e99acbd256a

cc @benakansara @baileycash-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368719","number":368719,"commit":"3d0b25be9b00bbdf0eb688db4ec5337d367b3023"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368673","number":368673,"commit":"b13a985689f8021310e19a94e7e8a5340263651b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368054","number":368054,"commit":"8569255d0ae6c3785a39a96cddd543fe1734a744"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368009","number":368009,"commit":"f7b623310949af3d8a9f43ade6e22f9cb383781e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367443","number":367443,"commit":"1e0f8454e3833dd6d026eff222460feb368ea368"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367228","number":367228,"commit":"37cefd867bf2292b79d92e9ffc2f0e99acbd256a"}],"number":368719}
buildkite-pr-comment-->

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#9980](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9980)
[✅] x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/detection_engine/exceptions/operators_data_types/float/basic_license_essentials_tier/configs/serverless.config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F243792%2Fhead)

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

baileycash-elastic committed:
Merge branch 'main' into feat/edit-workflow-tags

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369283)
* Commit: a47954bab279c95e11b33d93baccc66238577082

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/38106c26e880d5ff0faf3a27c5582da5b83b9949) | [after](https://github.com/elastic/kibana/commit/a47954bab279c95e11b33d93baccc66238577082) | diff |
| --- | --- | --- | --- |
| `cases` | 1200 | 1208 | +8 |
| `embeddableAlertsTable` | 564 | 573 | +9 |
| `ml` | 2818 | 2827 | +9 |
| `observability` | 1618 | 1627 | +9 |
| `securitySolution` | 8465 | 8473 | +8 |
| `triggersActionsUi` | 1094 | 1103 | +9 |
| total |  |  | +52 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/38106c26e880d5ff0faf3a27c5582da5b83b9949) | [after](https://github.com/elastic/kibana/commit/a47954bab279c95e11b33d93baccc66238577082) | diff |
| --- | --- | --- | --- |
| `cases` | 1.4MB | 1.4MB | +12.1KB |
| `embeddableAlertsTable` | 1001.0KB | 1013.7KB | +12.7KB |
| `ml` | 5.6MB | 5.6MB | +12.6KB |
| `observability` | 1.7MB | 1.7MB | +12.8KB |
| `securitySolution` | 11.1MB | 11.1MB | +12.1KB |
| `triggersActionsUi` | 1.5MB | 1.5MB | +12.8KB |
| total |  |  | +75.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/38106c26e880d5ff0faf3a27c5582da5b83b9949) | [after](https://github.com/elastic/kibana/commit/a47954bab279c95e11b33d93baccc66238577082) | diff |
| --- | --- | --- | --- |
| `embeddableAlertsTable` | 5.8KB | 5.8KB | +1.0B |
| `observability` | 95.7KB | 95.7KB | -1.0B |
| total |  |  | -0.0B |



### History
* :broken_heart: [Build #369238](https://buildkite.com/elastic/kibana-pull-request/builds/369238) failed ec64aa82066b92a497e027dbeb5380eabad6c4c3
* :yellow_heart: [Build #368719](https://buildkite.com/elastic/kibana-pull-request/builds/368719) was flaky 3d0b25be9b00bbdf0eb688db4ec5337d367b3023
* :broken_heart: [Build #368673](https://buildkite.com/elastic/kibana-pull-request/builds/368673) failed b13a985689f8021310e19a94e7e8a5340263651b
* :yellow_heart: [Build #368054](https://buildkite.com/elastic/kibana-pull-request/builds/368054) was flaky 8569255d0ae6c3785a39a96cddd543fe1734a744
* :green_heart: [Build #368009](https://buildkite.com/elastic/kibana-pull-request/builds/368009) succeeded f7b623310949af3d8a9f43ade6e22f9cb383781e
* :broken_heart: [Build #367443](https://buildkite.com/elastic/kibana-pull-request/builds/367443) failed 1e0f8454e3833dd6d026eff222460feb368ea368

cc @benakansara @baileycash-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369283","number":369283,"commit":"a47954bab279c95e11b33d93baccc66238577082"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369238","number":369238,"commit":"ec64aa82066b92a497e027dbeb5380eabad6c4c3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368719","number":368719,"commit":"3d0b25be9b00bbdf0eb688db4ec5337d367b3023"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368673","number":368673,"commit":"b13a985689f8021310e19a94e7e8a5340263651b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368054","number":368054,"commit":"8569255d0ae6c3785a39a96cddd543fe1734a744"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368009","number":368009,"commit":"f7b623310949af3d8a9f43ade6e22f9cb383781e"}],"number":369283}
buildkite-pr-comment-->

benakansara committed:
update header

benakansara committed:
Merge branch 'main' into feat/edit-workflow-tags

cnasikas reviewed (commented):
LGMT! I found a bug. If you run Kibana with `--run-examles` and you create an Always firing rule, and then you go to the detailed page of the rule you see the following:

https://github.com/user-attachments/assets/8b4463d2-5436-452a-ab87-b13ca4fcaad8

cnasikas commented on src/platform/packages/shared/response-ops/alerts-table/components/tags/use_tags_action.tsx:
```diff
@@ -0,0 +1,92 @@
+      }, {} as Record<string, AlertIdentifier[]>);
+
+      try {
+        await Promise.all(
```
Done: https://github.com/elastic/kibana/issues/245378

baileycash-elastic committed:
simplify tests, address feedback

benakansara committed:
fix close flyout on error

cnasikas reviewed (approved):
LGTM! Thanks for your patience with my comments!

benakansara committed:
Merge branch 'main' into feat/edit-workflow-tags

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#9998](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9998)
[✅] x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group4/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F243792%2Fhead)
